sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -178,13 +195,102 @@ internal class ViewAnnotationManagerImpl(
 
     if (viewAnnotationOptions.isEmpty()) return null
     val coordinates = coordinatesFromAnnotations(viewAnnotationOptions)
-    val paddings = calculateEdgeInsets(viewAnnotationOptions, edgeInsets)
-    return mapboxMap.cameraForCoordinates(
+    var cameraOptionForCoordinates = mapboxMap.cameraForCoordinates(
       coordinates,
-      paddings,
+      EdgeInsets(0.0, 0.0, 0.0, 0.0),
       bearing,
       pitch
     )
+
+    var isCorrectBound = false
+    // keep north, east, west and southmost Annotation and its frame to adjust paddings for zoom.
+    var north: Pair<ViewAnnotationOptions, Rect?>? = null
+    var east: Pair<ViewAnnotationOptions, Rect?>? = null
+    var west: Pair<ViewAnnotationOptions, Rect?>? = null
+    var south: Pair<ViewAnnotationOptions, Rect?>? = null
+
+    while (!isCorrectBound) {
+      val zoom = cameraOptionForCoordinates.zoom
+      isCorrectBound = true
+      viewAnnotationOptions.forEach { options ->
+        val frame = getViewAnnotationOptionsFrame(options)
+        val annotationBounds = calculateCoordinateBoundForAnnotation(options, zoom)
+        if (north == null || calculateCoordinateBoundForAnnotation(north!!.first, zoom).north() < annotationBounds.north()) {
+          north = Pair(options, frame)
+          isCorrectBound = false
+        }
+        if (east == null || calculateCoordinateBoundForAnnotation(east!!.first, zoom).east() < annotationBounds.east()) {
+          east = Pair(options, frame)
+          isCorrectBound = false
+        }
+        if (south == null || calculateCoordinateBoundForAnnotation(south!!.first, zoom).south() > annotationBounds.south()) {
+          south = Pair(options, frame)
+          isCorrectBound = false
+        }
+        if (west == null || calculateCoordinateBoundForAnnotation(west!!.first, zoom).west() > annotationBounds.west()) {
+          west = Pair(options, frame)
+          isCorrectBound = false
+        }
+      }
+
+      // adding extra checks for nullability (this shouldn't execute normally as we are checking nullability in loop of viewannotations).
+      if (north == null || east == null || south == null || west == null) {
+        logW(TAG, "ViewAnnotation options framing is null. Returning null camera option")
+        return null
+      }
+
+      val coordinateBoundsForCamera = CoordinateBounds(
+        Point.fromLngLat(west!!.coordinate().longitude(), south!!.coordinate().latitude()),
+        Point.fromLngLat(east!!.coordinate().longitude(), north!!.coordinate().latitude())
+      )
+
+      val paddings = EdgeInsets(
+        (edgeInsets?.top ?: 0).toDouble().plus(abs((north!!.second?.top ?: 0.0).toDouble())),
+        (edgeInsets?.left ?: 0).toDouble().plus(abs((west!!.second?.left ?: 0.0).toDouble())),
+        (edgeInsets?.bottom ?: 0).toDouble().plus(abs((south!!.second?.bottom ?: 0.0).toDouble())),
+        (edgeInsets?.right ?: 0).toDouble().plus(abs((east!!.second?.right ?: 0.0).toDouble()))
```

Comment:
Reads hardly, maybe 
```suggestion
        (edgeInsets?.right ?: 0).toDouble() + abs((east!!.second?.right ?: 0.0).toDouble())
```
?